api/v2/checluster_types.go

```diff
@@ -72,6 +73,15 @@ type CheClusterDevEnvironments struct {
 	// Trusted certificate settings.
 	// +optional
 	TrustedCerts TrustedCerts `json:"trustedCerts,omitempty"`
+	// The default editor to workspace create with. It could be a plugin ID or a URI.
+	// The plugin ID must have `publisher/plugin/version` format.
+	// The URI must start from `http://` or `https://`.
+	// +optional
+	DefaultEditor string `json:"defaultEditor,omitempty"`
+	// Default components applied to DevWorkspaces.
+	// These default components are meant to be used when a Devfile, that does not contain any components.
+	// +optional
+	DefaultComponents []devfile.Component `json:"defaultComponents,omitempty"`
```

Comment:
@l0rd please, take a look at the final structure we are planning to have in place:
```
devEnvironments:
  (...)
  defaultComponents [ ] 
  defaultEditor: ""
```
Any concerns?